src/contrib/cluster/Akka.DistributedData/Serialization/ReplicatedDataSerializer.cs

```diff
@@ -992,7 +992,10 @@ private ORDictionary.IDeltaGroupOp ORDictionaryDeltaGroupFromProto(Proto.Msg.ORM
 
                 if (entry.Value != null)
                 {
-                    return (k, (TValue)_ser.OtherMessageFromProto(entry.Value));
+                    // Bug #4367 throws here, value is ORDictionary.DeltaValue,
+                    // which can't be cast to ORSet
+                    var value = _ser.OtherMessageFromProto(entry.Value);
```

Comment:
The bug turned out to be simpler than I thought - just a casting problem with how we handle `ORDictionary.DeltaGroup`s. I moved the casting out of this method and instead dealt with it at the point where we reconstitute the `DeltaGroup`. Looks like this turned all of the tests green,